lib/puppet/pops/types/types.rb

```diff
@@ -2799,6 +2800,12 @@ def really_instance?(o, guard = nil)
     resolved_type.really_instance?(o, guard)
   end
 
+  # @return `nil` to prevent serialization of the type_expr used when first initializing this instance
+  # @api private
+  def type_expr
+    nil
```

Comment:
I got that - should we support volatile (non serialized) attributes in Pcore perhaps? Would that solve this case?